src/base/gps_test.cc

```diff
@@ -165,3 +165,133 @@ BOOST_AUTO_TEST_CASE(TestXYZToEllToXYZ_WGS84) {
+
+  GPSTransform gps_tform(GPSTransform::WGS84);
+
+  const auto xyz = gps_tform.EllToXYZ(ell);
+
+  // Get lat0, lon0 origin from ref
+  const auto ori_ell = gps_tform.XYZToEll({ref_xyz[0]})[0];
+
+  // Get ENU from ECEF ref
+  const auto ref_enu = gps_tform.XYZToENU(ref_xyz, ori_ell(0), ori_ell(1));
+
+  // Get ENU from ECEF
+  const auto enu = gps_tform.XYZToENU(xyz, ori_ell(0), ori_ell(1));
+
+  for (size_t i = 0; i < ell.size(); ++i) {
+    BOOST_CHECK(std::abs(enu[i](0) - ref_enu[i](0)) < 1e-8);
+    BOOST_CHECK(std::abs(enu[i](1) - ref_enu[i](1)) < 1e-8);
+    BOOST_CHECK(std::abs(enu[i](2) - ref_enu[i](2)) < 1e-8);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(TestENUToEllWGS84) {
+  std::vector<Eigen::Vector3d> ref_ell;
+  ref_ell.emplace_back(48 + 8. / 60 + 51.70361 / 3600,
+                       11 + 34. / 60 + 10.51777 / 3600, 561.1851);
+  ref_ell.emplace_back(48 + 8. / 60 + 52.40575 / 3600,
+                       11 + 34. / 60 + 11.77179 / 3600, 561.1509);
+
+  std::vector<Eigen::Vector3d> xyz;
+  xyz.emplace_back(4.177239709042750e6, 0.855153779923415e6,
+                   4.728267404769168e6);
+  xyz.emplace_back(4.177218660452103e6, 0.855175931344048e6,
+                   4.728281850382507e6);
+
+  GPSTransform gps_tform(GPSTransform::WGS84);
+
+  // Get lat0, lon0 origin from ref
+  const auto ori_ell = gps_tform.XYZToEll(xyz);
+  const double lat0 = ori_ell[0](0);
+  const double lon0 = ori_ell[0](1);
+  const double alt0 = ori_ell[0](2);
+
+  // Get ENU from ECEF
+  const auto enu = gps_tform.XYZToENU(xyz, lat0, lon0);
+
+  const auto xyz_enu = gps_tform.ENUToXYZ(enu, lat0, lon0, alt0);
+
+  // Get Ell from ENU
+  const auto ell = gps_tform.ENUToEll(enu, lat0, lon0, alt0);
+
+  for (size_t i = 0; i < ell.size(); ++i) {
+    BOOST_CHECK(std::abs(ell[i](0) - ref_ell[i](0)) < 1e-5);
+    BOOST_CHECK(std::abs(ell[i](1) - ref_ell[i](1)) < 1e-5);
+    BOOST_CHECK(std::abs(ell[i](2) - ref_ell[i](2)) < 1e-5);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(TestENUToXYZ) {
+  std::vector<Eigen::Vector3d> ell;
+  ell.emplace_back(48 + 8. / 60 + 51.70361 / 3600,
+                   11 + 34. / 60 + 10.51777 / 3600, 561.1851);
+  ell.emplace_back(48 + 8. / 60 + 52.40575 / 3600,
+                   11 + 34. / 60 + 11.77179 / 3600, 561.1509);
+  std::vector<Eigen::Vector3d> ref_xyz;
+  ref_xyz.emplace_back(4.177239709042750e6, 0.855153779923415e6,
+                       4.728267404769168e6);
+  ref_xyz.emplace_back(4.177218660452103e6, 0.855175931344048e6,
+                       4.728281850382507e6);
+
+  GPSTransform gps_tform(GPSTransform::WGS84);
+
+  // Get lat0, lon0 origin from Ell
+  const double lat0 = ell[0](0);
+  const double lon0 = ell[0](1);
+  const double alt0 = ell[0](2);
+
+  // Get ENU from Ell
+  const auto enu = gps_tform.EllToENU(ell);
+
+  // Get XYZ from ENU
+  const auto xyz = gps_tform.ENUToXYZ(enu, lat0, lon0, alt0);
+
+  for (size_t i = 0; i < ell.size(); ++i) {
+    BOOST_CHECK(std::abs(xyz[i](0) - ref_xyz[i](0)) < 1e-8);
```

Comment:
Could be BOOST_CHECK(xyz[i].isApprox(ref_xyz[i], 1e-8))

Reply:
Thanks, I ended up doing this for all test functions in gps_test.cc